cmd/spx/template/game.js

```diff
@@ -32,27 +32,92 @@ class GameApp {
     }
 
     StartProject() {
-        this.installProject();
+        this.isEditor = true
+        if (this.editor != null) {
+            return new Promise((resolve, reject) => {
+                let error = new Error("project already loaded!");
+                console.log(error)
+                reject(error)
+            });
+        }
+        let promise = new Promise((resolve, reject) => {
+            this.startProjectResolve = resolve
+            this.startProject();
+        });
+        this.startProjectPromise = promise
+        return promise
     }
 
     async UpdateProject(newData, addInfos, deleteInfos, updateInfos) {
-        console.log("UpdateProject", this.oldData, newData)
-        if (addInfos == undefined) {
-            const diffInfos = await this.getZipDiffInfos(this.oldData, newData);
-            addInfos = diffInfos.addInfos;
-            deleteInfos = diffInfos.deleteInfos;
-            updateInfos = diffInfos.updateInfos;
+        if (this.startProjectPromise != null) {
+            await this.startProjectPromise
+        }
+        if (this.updateProjectPromise != null) {
+            await this.updateProjectPromise
+        }
+        let promise = new Promise(async (resolve, reject) => {
```

Comment:
If we call `UpdateProject` multiple times:

```js
UpdateProject(...)
// While the 1st `UpdateProject` not finished, call `UpdateProject` very quickly for another 2 times
UpdateProject(...)
UpdateProject(...)
```

The 2rd & 3rd `UpdateProject` will both wait for the 1st to finish, while the 3rd will not wait for the 2rd to finish. As a result, the 2rd & 3rd `updateProject` will run "concurrently", which I'm not sure if is expected.

Reply:
yes, i will fixed it in the next few commits

Reply:
this has been resove by https://github.com/goplus/spx/pull/378